superset-frontend/src/features/datasets/AddDataset/LeftPanel/index.tsx

```diff
@@ -239,140 +156,39 @@ export default function LeftPanel({
     }
   }, [setDatabase]);
 
-  useEffect(() => {
-    if (loadTables) {
-      const params = rison.encode({
-        force: refresh,
-        schema_name: encodedSchema,
-      });
-
-      const endpoint = `/api/v1/database/${dataset?.db?.id}/tables/?q=${params}`;
-      getTablesList(endpoint);
-    }
-  }, [loadTables, dataset?.db?.id, encodedSchema, getTablesList, refresh]);
-
-  useEffect(() => {
-    if (resetTables) {
-      setTableOptions([]);
-      setResetTables(false);
-    }
-  }, [resetTables]);
-
-  const filteredOptions = tableOptions.filter(option =>
-    option?.value?.toLowerCase().includes(searchVal.toLowerCase()),
-  );
-
-  const Loader = (inline: string) => (
-    <div className="loading-container">
-      <Loading position="inline" />
-      <p>{inline}</p>
-    </div>
+  const datasetNamesSet = datasetNames?.join(divider);
```

Comment:
Can you use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) here?

Reply:
The purpose of using concatenated string is avoiding the re-rendering by the reference differences because it will shallowly comparing only values given the different reference of datasetNames passing from the property